### PR TITLE
VideoPress: disable privacy button when updating video privacy

### PR DESCRIPTION
**Pull request by retrofox**
Fixes #

#### Changes proposed in this Pull Request:

* VideoPress:
  * Reduxify video privacy data: selector, reduce the data, expose the data from useVideo() hook, etc
  * Disable privacy button when the video privacy is updating

#### Testing instructions:

* Go to VideoPress dashboard
* Ensure you have a VideoPerss video
* Change the video privacy
* Confirm the button is disabled when updating the video privacy

https://user-images.githubusercontent.com/77539/194386864-d1d3cca1-dda4-417d-a7a2-dadc1b1ac645.mov

### CODE REVIEW

File: projects/packages/videopress/src/client/admin/hooks/use-video/index.ts [hunk 1]
```diff
@@ -23,6 +23,11 @@ export default function useVideo( id: number | string ) {
 		[ id ]
 	);
 
+	const metadata = useSelect(
+		select => ( select( STORE_ID ) as VideopressSelectors ).getVideoStateMetadata( id ),
+		[]
+	);
+
```
Comment (dhasilva) [inline projects/packages/videopress/src/client/admin/hooks/use-video/index.ts:30]: :100:

File: projects/packages/videopress/src/client/state/actions.js [hunk 2]
```diff
@@ -74,7 +80,7 @@ const updateVideoPrivacy = ( id, level ) => async ( { dispatch } ) => {
 
 	// Let's be optimistic and update the UI right away.
 	// @todo: Add a loading state to the state/UI.
-	dispatch.setVideo( {
+	dispatch.setVideoPrivacy( {
```
Comment (dhasilva) [inline projects/packages/videopress/src/client/state/actions.js:82]: The `@todo` above is addressed in this PR

Comment (retrofox) [reply projects/packages/videopress/src/client/state/actions.js:82]: yes :'(

Comment (retrofox) [reply projects/packages/videopress/src/client/state/actions.js:82]: let me remove it in a follow-up. I wouldn't like to restart the tests just for this.

File: projects/packages/videopress/src/client/state/reducers.js [hunk 3]
```diff
@@ -124,6 +126,72 @@ const videos = ( state, action ) => {
...
+			return {
+				...state,
+				_meta: {
+					...state._meta,
+					items: {
+						..._metaItems,
+						[ id ]: {
+							..._metaVideo,
+							isUpdatingPrivacy: false,
+							hasBeenUpdatedPrivacy: true,
```
Comment (dhasilva) [inline projects/packages/videopress/src/client/state/reducers.js:188]: Shouldn't `prevPrivacySetting` be set to `null`?

Comment (retrofox) [reply projects/packages/videopress/src/client/state/reducers.js:188]: yes, makes sense. Updating...

**Review by dhasilva — APPROVED** [review]
Looking good! :rocket:

**Pull request merged**